### PR TITLE
ROC-2294 Duplicate claim error handling

### PR DESCRIPTION
**Pull request by dharmendrak**

### CODE REVIEW

File: src/main/common/router/routablePath.ts [hunk 1]
```diff
@@ -23,4 +25,28 @@ export class RoutablePath {
     const viewPath: string = split.slice(2).join('/')
     return `${featureName}/views/${viewPath}`
   }
+
+  evaluateUri (substitutions: { [key: string]: string }): string {
```
Comment (timja-kainos) [inline src/main/common/router/routablePath.ts]: missing tests for this? (we really need to pull this sort of stuff out to a common module)

File: src/main/features/claim/routes/pay-by-account.ts [hunk 2]
```diff
@@ -8,10 +9,52 @@ import ErrorHandling from 'common/errorHandling'
...
+
+  let claimStatus: boolean
+  try {
+    claimStatus = await ClaimStoreClient.retrieveByExternalId(externalId)
+      .then(() => true)
+  } catch (err) {
+    if (err.toString().includes('Claim not found by external id')) {
+      claimStatus = false
+    } else {
+      logError(res.locals.user.id, `There is problem retrieving claim from claim store externalId: ${externalId},`)
```
Comment (timja-kainos) [inline src/main/features/claim/routes/pay-by-account.ts]: a problem

File: src/main/features/claim/routes/pay-by-account.ts [hunk 3]
```diff
@@ -8,10 +9,52 @@ import ErrorHandling from 'common/errorHandling'
...
+
+async function saveClaimHandler (res, next) {
+  const externalId = res.locals.user.legalClaimDraft.externalId
+
+  let claimStatus: boolean
+  try {
+    claimStatus = await ClaimStoreClient.retrieveByExternalId(externalId)
+      .then(() => true)
+  } catch (err) {
+    if (err.toString().includes('Claim not found by external id')) {
```
Comment (timja-kainos) [inline src/main/features/claim/routes/pay-by-account.ts]: should be able to check status code, not relying on the message, (I know we do this in T1 but we hacked it together really quickly)

Comment (dharmendrak) [reply src/main/features/claim/routes/pay-by-account.ts]: done

File: src/main/features/claim/routes/pay-by-account.ts (hunk 2 above)
Comment (dharmendrak) [reply src/main/features/claim/routes/pay-by-account.ts]: done

File: src/main/common/router/routablePath.ts (hunk 1 above)
Comment (dharmendrak) [reply src/main/common/router/routablePath.ts]: written tests

File: src/test/http-mocks/claim-store.ts [hunk 4]
```diff
@@ -44,7 +44,7 @@ export function resolveRetrieveClaimByExternalId (claimOverride?: object) {
 export function rejectRetrieveClaimByExternalId (reason: string) {
   mock(`${serviceBaseURL}/claims`)
     .get(new RegExp('/[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}'))
-    .reply(HttpStatus.INTERNAL_SERVER_ERROR, reason)
+    .reply(HttpStatus.NOT_FOUND, reason)
```
Comment (timja-kainos) [inline src/test/http-mocks/claim-store.ts]: undo this change

File: src/test/http-mocks/claim-store.ts [hunk 5]
```diff
@@ -44,7 +44,7 @@ export function resolveRetrieveClaimByExternalId (claimOverride?: object) {
 export function rejectRetrieveClaimByExternalId (reason: string) {
```
Comment (timja-kainos) [inline src/test/http-mocks/claim-store.ts]: rename this to have NotFound in it
